platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/Configurator.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
```

Comment:
❗️Since this is meant to be an external API (at least based on the package name), we should add a description explaining what this interface is and when and how it should be used. For instance, it may be useful to clarify that an "operation" is a call to one of the other methods and that `commit` should be called once all the desired method have been called.